rootston/output.c

```diff
@@ -696,14 +696,10 @@ static void damage_from_surface(struct wlr_surface *surface, int sx, int sy,
 	int center_x = box.x + box.width/2;
 	int center_y = box.y + box.height/2;
 
-	enum wl_output_transform transform =
-		wlr_output_transform_invert(surface->current.transform);
-
 	pixman_region32_t damage;
 	pixman_region32_init(&damage);
-	pixman_region32_copy(&damage, &surface->buffer_damage);
-	wlr_region_transform(&damage, &damage, transform,
-		surface->current.buffer_width, surface->current.buffer_height);
+	wlr_surface_get_effective_damage(surface, &damage);
+
 	wlr_region_scale(&damage, &damage,
 		wlr_output->scale / (float)surface->current.scale);
```

Comment:
You need to remove `surface->current.scale` here.

Can you test with `scale=2` in your config file?